src/docs/arch-compact.dox

```diff
@@ -0,0 +1,57 @@
+/*! @arch_page arch-compact Compaction
+
+Compaction is a cooperative process between the @ref arch-btree "Btree layer" and
+the @ref arch-block "Block Manager" to reduce the on-disk footprint of WT tables. The compaction
+process is initiated by user application by calling \c WT_SESSION::compact method. Overall, the
+compaction process involves rewriting disk blocks from the end of the on-disk file to the start of
+the file, hence giving an opportunity to reclaim the space from the end of file through truncation
+and effectively reducing the file system reported file size. Note that the compaction is a
+"best effort" process and does not guarantees reduction in file size.
+
+Before understanding compaction process, we need to understand how disk blocks are referenced by the
+checkpoints. Each checkpoint references a certain set of disk blocks for a table.. When a dirty
+btree page is reconciled, a new disk block is assigned for the page and a new checkpoint starts
+referencing the new block used for the page. Once the checkpoint referencing the old block is deleted,
+the old disk block becomes available for reuse. The aim of compaction is to copy over the data from
+the blocks at the end of file to the blocks in the start of the file that are available for reuse.
+
+The compaction process starts with a system-wide checkpoint. There are potentially many dirty blocks
+in the cache, and the intention here is to write them out and then discard previous checkpoints so
+that there are as many blocks as possible on the file's "available for reuse" list when the compaction
+starts.  The compaction logic then starts examining the files that need to be compacted. Compacting
+the object is done 10% at a time, that is, compaction tries to move blocks from the last 10% of the
+file into the beginning of the file (the 10% is hard coded in the block manager).  The reason for
```

Comment:
Do you think the following information would be useful here?
`We then compact the high-level object, ignoring files smaller than 1MB. We compute the available bytes in the initial 80% and 90% of the file. If we can’t recover at least 1MB, the file is skipped. If at least 20% of the file is available in the first 80% of the file, we'll try compaction on the last 20% of the file. Else if at least 10% of the total file is available in the first 90% of the file, we'll try compaction on the last 10% of the file. Else the file is skipped.`

Reply:
Good point. Can we use the flowchart I created for the knowledge-sharing session?
Link to the presentation https://docs.google.com/presentation/d/17_vNBtBie3fm4kYRRS6CbBeMRRcTD5a7jSeQ34FDDw8/edit#slide=id.gf085a5dda9_0_18

Reply:
I will go for the text option as making the flow chart might be a more complex task as we need to draw diagrams suing UML drawer.

Reply:
Added more details in 3d77c6f.

Reply:
Added more details in 3d77c6f.